src/compiler/checker.ts

```diff
@@ -15905,7 +15901,7 @@ namespace ts {
                 const symbol = isNonConstructorObject ? target.symbol : undefined;
                 if (symbol) {
                     if (contains(symbolStack, symbol)) {
-                        inferenceIncomplete = true;
+                        inferencePriority = -1;

```

Comment:
I can't tell if `-1` is being used here because it's lower than anything else, has all bits set, or both. A named enum member with the intended semantics of this would be clearer.

Reply:
It's because `-1` is less than any normal priority value. I'll give it a symbolic name.